### PR TITLE
Optimize the plots sections to reduce the number of useless re-renderings

### PR DESCRIPTION
**Pull request by sroy3**
Part of #3334 

### Demo before

https://user-images.githubusercontent.com/3683420/221030411-bd441d74-d36f-4d4d-ae47-5aad60815935.mov

### Demo now

https://user-images.githubusercontent.com/3683420/221030697-e30fe089-db23-4776-a15f-d382147f3efc.mov

Not a big improvement, but this should add up with the number of plots and it should make the code a little easier to read.

### CODE REVIEW

File: webview/src/plots/components/templatePlots/TemplatePlots.tsx [hunk 1]
```diff
@@ -43,29 +35,30 @@ export const TemplatePlots: React.FC = () => {
...
+      })),
+      type: MessageFromWebviewType.REORDER_PLOTS_TEMPLATES
+    })
+  }, [])
+
+  const setSections = useCallback(
+    (sections: PlotGroup[]) => {
+      /* Although the following dispatch duplicates the work the reducer will do when the state returns 
+         from the extension, this is necessary to not see any flickering in the order as the returned state 
+         sometimes takes a while to come back */
```
Comment (mattseddon) [inline webview/src/plots/components/templatePlots/TemplatePlots.tsx:56]: 🎉

**Review by mattseddon — APPROVED** [review]

**Comment by mattseddon** [comment]
[Q] Can we automate re-render performance testing in some way?

**Comment by sroy3** [comment]
> [Q] Can we automate re-render performance testing in some way?
> 
> Very quick look around found this: [testing-library/react-hooks-testing-library#135](https://github.com/testing-library/react-hooks-testing-library/issues/135)

I'll take a little bit of time to research this today. I think this could be very useful in the future. Even with virtualization, there is a lot of data rendered at once. Making sure we do not worsen re-rendering performance should be on own mind when developing in the plots webview (experiments table as well).

**Comment by sroy3** [comment]
[Profiler API](https://reactjs.org/docs/profiler.html) is probably the best tool for testing re-renders. The problem is that we don't want to test the number of re-renders for an isolated component and we do not want to profile simply the `<App />` component either. What would be best, would be being able to wrap a certain component within the whole `<App />` component. I'll see if it's possible to create a test helper method to solve that problem.

**Pull request merged**